src/main/cpp/src/DecimalUtilsJni.cpp

```diff
@@ -19,8 +19,13 @@
 
 extern "C" {
 
-JNIEXPORT jlongArray JNICALL Java_com_nvidia_spark_rapids_jni_DecimalUtils_multiply128(
-  JNIEnv* env, jclass, jlong j_view_a, jlong j_view_b, jint j_product_scale)
+JNIEXPORT jlongArray JNICALL
+Java_com_nvidia_spark_rapids_jni_DecimalUtils_multiply128(JNIEnv* env,
+                                                          jclass,
+                                                          jlong j_view_a,
+                                                          jlong j_view_b,
+                                                          jint j_product_scale,
+                                                          bool cast_interim_result)
```

Comment:
The question now is, since this new parameter is added to avoid a known bug, will it be turned on all the time to fix the bug, or will it be turned on/off depending on the situations? If so then when it is turned on?

Reply:
The new parameter is only set to false for Spark versions for which the multiplication bug was fixed. Namely Spark versions 3.4.2, 4.0.0, 3.5.1